### PR TITLE
✨Add focus and blur events

### LINKED ISSUE #11248: Support focus and blur events
## What's the issue?

__Feature request__: support events, at least on input elements (`<input>`, `<select>`, `<textarea>`, etc.), that fire when the element gains or loses focus, similar to the JavaScript `onfocus` and `onblur` event.

A sample use case is having an initially-hidden submit button, which only gets displayed when the text field is non-empty or focused and will get hidden when the input field is empty and loses focus.

Currently, the focus event can be simulated by using `tap` event, but that only works for mouse and not the tab key. Also, If the user presses the mouse inside the input but releases it outside, the action is not considered a `tap`, but the input does gain focus. The blur event is pretty hard to simulate, as the focus can be lost to any other elements.

## How do we reproduce the issue?

The usage can be something like this:

```html
<form ...>
  <input type="text" on="change:AMP.setState({empty: !event.value});focus:AMP.setState({focused: true});blur:AMP.setState({focused: false})">
  <button id="submit-button" class="jfk-button jfk-button-action" [hidden]="!empty || focused">
    Submit
  </button>
</form>

```

### PR DESCRIPTION
**Pull request by kevinkassimo**
Adds focus and blur events. The handlers are using capture since these event never bubbles.

Closes #11248

### CODE REVIEW

**Comment by kevinkassimo** [comment]
@aghassemi Could you review this? Thanks!

File: src/service/action-impl.js [hunk 1]
```diff
@@ -332,6 +334,16 @@ export class ActionService {
         const element = dev().assertElement(event.target);
         this.trigger(element, name, event, ActionTrust.HIGH);
       });
+    } else if (name == 'focus') {
+      this.root_.addEventListener(name, event => {
+        const element = dev().assertElement(event.target);
+        this.trigger(element, name, event, ActionTrust.HIGH);
```
Comment (aghassemi) [inline src/service/action-impl.js]: There is one tricky thing we need to figure out:
`focus`/ `blue` can happen on elements without user interaction in certain cases such as using '#idOfInput` in Url or using `autofocus` attribute . We should either not fire the events in those cases or fire it with "Low-trust" (I vote for former).

I am not sure how we can detect it, I believe the `event` target when focus is done via user-action would have something different than when it is automatic. Some investigation is needed.

Comment (nainar) [reply src/service/action-impl.js]: Pinging @dvoytenko to take a look here.

**Comment by SAGHEERSAYONE19** [comment]
Hi I have an otp (6 digit) in my form, and I want to focus on next input,what should be done
```<div class="form_input"><input name="otp" type="number" class="input_form"></div>
                           <div class="form_input"><input name="otp" type="number" id="first" class="input_form"></div>
                           <div class="form_input"><input name="otp" type="number" id="second" class="input_form"></div>
                           <div class="form_input"><input name="otp" type="number" id="third" class="input_form"></div>
                           <div class="form_input"><input name="otp" type="number" id="fourth" class="input_form"></div>
                           <div class="form_input"><input name="otp" type="number" id="fifth" class="input_form"></div>```
This is my code

**Pull request closed without merging**